daemon/cmd/status.go

```diff
@@ -388,8 +388,20 @@ func (c *clusterNodesClient) NodeAdd(newNode nodeTypes.Node) error {
 
 func (c *clusterNodesClient) NodeUpdate(oldNode, newNode nodeTypes.Node) error {
 	c.Lock()
-	c.NodesAdded = append(c.NodesAdded, newNode.GetModel())
-	c.NodesRemoved = append(c.NodesRemoved, oldNode.GetModel())
+	// If the node is on the added list, just update it
+	found := -1
+	for i, added := range c.NodesAdded {
+		if added.Name == newNode.Fullname() {
+			found = i
```

Comment:
Maybe `found` could be `newNodeIdx`? ~~Also possible to break here.~~ Saw second commit, could you squash them?

Reply:
I thought it would be better to separate the concerns, but yes I can squash them into one commit.

Reply:
Ah I see, I had missed that you also changed `NodeDelete`'s implementation. Either way LGTM then!